util/src/main/java/org/folio/reservoir/util/readstream/XmlFixerMapper.java

```diff
@@ -165,47 +165,61 @@ public void push(Buffer buffer) {
 
   private void skipByte(Buffer input) {
     result.appendBuffer(input, tail, front - tail);
-    addFix();
-  }
-
-  private void addFix() {
     tail = front + 1;
     result.appendString(REPLACEMENT_CHAR);
     numberOfFixes++;
   }
 
   private boolean handleEntity(Buffer input) {
-    if (front == input.length() - 1) {
-      incomplete(input);
-      return true;
-    }
-    if (input.getByte(front + 1) != '#') {
-      return false;
-    }
-    int j;
-    for (j = front + 2; j < input.length(); j++) {
-      if (!isAscii(input.getByte(j)) || input.getByte(j) == ';') {
+    int j = front + 1;
+    while (true) {
+      if (j >= input.length()) {
+        incomplete(input);
+        return true;
+      }
+      byte b = input.getByte(j);
+      if (b == ';') {
         break;
       }
+      if (!isAscii(b) || b <= 32) {
+        return false;
+      }
+      j++;
     }
-    if (j == input.length()) {
-      incomplete(input);
-      return true;
+    int skip = 0;
+    if (input.getByte(front + 1) == '"') {
```

Comment:
I think we should cover the general case and skip up to 3 chars like we do for the UTF8 sequence if they dont' look like a character reference (&#nnnn; or &#xhhhh;) or one of the built-in XML entities (quot, amp, apos, lt, gt).

Reply:
Ok

Reply:
ok. that's more work. Will take probably most of today.

Reply:
Alternatively, we can check if  the char following & is alphanumeric [a-zA-Z0-9]. Advantage is that it would support custom entities, disadvantage is that it's very unlikely that documents would ever use custom entities and it's probably some HTML artifact that will cause the XML parsing to fail. Whatever is easier.

Reply:
I'm on it.  There are more cases where needs fixing,.  eg bad numeric entities that would just go through.

I'll add a check for the 5 standard entities. I don't think we should accept anything else.

Reply:
Agree